From 001717f424e88d7bfa6ba07cce129683817d3369 Mon Sep 17 00:00:00 2001
From: Ian Hinder <ian.hinder@aei.mpg.de>
Date: Fri, 21 Oct 2011 09:25:52 +0200
Subject: [PATCH] Change default of VECTORISE_INLINE to "no"

Most of the machines supported by SimFactory that use the Intel
compiler have to set this to "no" to avoid the compiler running out of
memory when compiling ML_BSSN.  We choose "no" as the default now to
avoid problems for users of machines not supported in SimFactory.
---
 README       |    9 ++++++---
 configure.sh |    2 +-
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/README b/README
index 8d0796f..cc50242 100644
--- a/README
+++ b/README
@@ -41,9 +41,12 @@ operations and corresponding vector-gather operations. This may be
 beneficial if unaligned load operations are slow, and if vector-gather
 operations are fast.
 
-VECTORISE_INLINE (default "yes"): Inline functions into the loop body
-as much as possible. (Disabling this may reduce code size, which can
-improve performance if the instruction cache is small.)
+VECTORISE_INLINE (default "no"): Inline functions into the loop body
+as much as possible.  This can cause some compilers to run out of
+memory for complex codes. (Enabling this may increase code size, which
+can degrade performance if the instruction cache is small, but may
+lead to increased performance due to omission of function-call
+overhead.)
 
 VECTORISE_STREAMING_STORES (default "yes"): Use streaming stores, i.e.
 use store operations that bypass the cache. (Disabling this produces
diff --git a/configure.sh b/configure.sh
index 83ae33a..dffbb92 100644
--- a/configure.sh
+++ b/configure.sh
@@ -79,7 +79,7 @@ esac
 case $(echo "x$VECTORISE_INLINE" | tr '[:upper:]' '[:lower:]') in
     (xyes) VECTORISE_INLINE=1 ;;
     (xno)  VECTORISE_INLINE=0 ;;
-    (x)    VECTORISE_INLINE=1 ;; # default
+    (x)    VECTORISE_INLINE=0 ;; # default
     (*)    echo "BEGIN ERROR"
            echo "Illegal value of option VECTORISE_INLINE"
            echo "END ERROR"
-- 
1.7.5.4

